src/internal/ppsutil/util.go

```diff
@@ -425,3 +429,38 @@ func GetWorkerPipelineInfo(pachClient *client.APIClient, env serviceenv.ServiceE
 
 	return pipelineInfo, nil
 }
+
+func FindPipelineSpecCommit(ctx context.Context, pfsServer pfsServer.APIServer, txnEnv transactionenv.TransactionEnv, pipeline string) (*pfs.Commit, error) {
+	var commit *pfs.Commit
+	if err := txnEnv.WithReadContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
```

Comment:
Very minor style idea (and I know this is a pure refactoring PR, so maybe it's better not to change anything at all), but if you wanted I think you could use a named return value, omit the declaration, and save a line e.g.:
```
if err := txnEnv.WithReadContext(ctx, func(txnCtx *txncontext.TransactionContext) (err error) {
	commit, err = FindPipelineSpecCommitInTransaction(txnCtx, pfsServer, pipeline, "")
	return;
}); err != nil {...
```

(per https://tour.golang.org/basics/7)